lib/ansible/module_utils/scaleway.py

```diff
@@ -21,6 +22,32 @@ def __init__(self, message):
         self.message = message
 
 
+# Specify a complete Link header, for validation purposes
+R_LINK_HEADER = r'''</[a-z]+\?page=[0-9]+&per_page=[0-9]+&>;
```

Comment:
Instead of building regexes to match the link parameters and extract them, it would be easier to use `parse_qs` which pulls values from valid query strings. For example:

```
>>> from ansible.module_utils.six.moves.urllib import parse
>>> query = parse.parse_qs('/foo/bar?page=6&per_page=20')
>>> print(query)
{'per_page': ['20'], '/foo/bar?page': ['6']}
>>> int(query['per_page'][0])
20
```

Or for more advanced parsing, you could use `urlparse` to remove the query string and path to use. 

```
>>> parse.urlparse('/foo/bar?page=6&per_page=20')
ParseResult(scheme='', netloc='', path='/foo/bar', params='', query='page=6&per_page=20', fragment='')
```